src/condor_io/condor_auth_ssl.cpp

```diff
@@ -1312,8 +1394,83 @@ int verify_callback(int ok, X509_STORE_CTX *store)
         X509_NAME_oneline( X509_get_issuer_name( cert ), data, 256 );
         dprintf( D_SECURITY, "  issuer   = %s\n", data );
         X509_NAME_oneline( X509_get_subject_name( cert ), data, 256 );
+	std::string dn(data);
         dprintf( D_SECURITY, "  subject  = %s\n", data );
         dprintf( D_SECURITY, "  err %i:%s\n", err, X509_verify_cert_error_string( err ) );
+
+		const SSL* ssl = (const SSL*)(*X509_STORE_CTX_get_ex_data_ptr)(store, (*SSL_get_ex_data_X509_STORE_CTX_idx_ptr)());
+		Condor_Auth_SSL::LastVerifyError *verify_ptr = (Condor_Auth_SSL::LastVerifyError *)(g_last_verify_error_index >= 0 ? (*SSL_get_ex_data_ptr)(ssl, g_last_verify_error_index) : nullptr);
+		if (verify_ptr) verify_ptr->m_skip_error = 0;
+
+		if (verify_ptr && ((err == X509_V_ERR_DEPTH_ZERO_SELF_SIGNED_CERT) ||
+			(err == X509_V_ERR_SELF_SIGNED_CERT_IN_CHAIN) ||
+			(err == X509_V_ERR_UNABLE_TO_GET_ISSUER_CERT_LOCALLY) ||
+			(err == X509_V_ERR_UNABLE_TO_GET_ISSUER_CERT)))
+		{
+			bool is_permitted;
+			std::string method, method_info;
+			auto encoded_cert = get_x509_encoded(cert);
+			bool is_ca_cert = (err == X509_V_ERR_UNABLE_TO_GET_ISSUER_CERT_LOCALLY) || (err == X509_V_ERR_UNABLE_TO_GET_ISSUER_CERT) ||
+				(err == X509_V_ERR_SELF_SIGNED_CERT_IN_CHAIN);
+
+			unsigned char         md[EVP_MAX_MD_SIZE];
```

Comment:
Generation of the fingerprint string should be deferred until we know we plan to use it.